mods/beds/api.lua

```diff
@@ -48,8 +48,8 @@ function beds.register_bed(name, def)
 			local under = pointed_thing.under
 			local node = minetest.get_node(under)
 			local udef = minetest.registered_nodes[node.name]
-			if udef and udef.on_rightclick and
-					not (placer and placer:get_player_control().sneak) then
+			if placer and udef and udef.on_rightclick and
```

Comment:
Even if placer is nil it should still call on_rightlick, just passing the placer (which is nil) through. Otherwise nil placers won't be able to use the item.